prepline_general/api/openapi.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any
+
+from fastapi import FastAPI
+from fastapi.openapi.utils import get_openapi
+
+
+def set_custom_openapi(app: FastAPI) -> None:
+    """Generate a custom OpenAPI schema for the app"""
+
+    def custom_openapi() -> dict[str, Any]:
+        if app.openapi_schema:
+            return app.openapi_schema
+        openapi_schema = get_openapi(
+            title=app.title,
+            version=app.version,
+            summary=app.summary,
+            description=app.description,
+            servers=app.servers,
+            routes=app.routes,
+            tags=app.openapi_tags,
+        )
+        _apply_customizations(openapi_schema)
+
+        app.openapi_schema = openapi_schema
+        return app.openapi_schema
+
+    app.openapi = custom_openapi  # type: ignore
+
+
+def _apply_customizations(openapi_schema: dict[str, Any]) -> None:
+    """Add customizations to the OpenAPI schema"""
+
+    # Add security
+    openapi_schema["security"] = [{"ApiKeyAuth": []}]
+
+    # Add retries
+    openapi_schema["x-speakeasy-retries"] = {
+        "strategy": "backoff",
+        "backoff": {
+            "initialInterval": 500,
+            "maxInterval": 60000,
+            "maxElapsedTime": 900000,
+            "exponent": 1.5,
+        },
+        "statusCodes": [
+            "5xx",
+        ],
+        "retryConnectionErrors": True,
+    }
+    # Path changes
+    # Update the $ref in paths
+    openapi_schema["paths"]["/general/v0/general"]["post"]["requestBody"]["content"][
+        "multipart/form-data"
+    ]["schema"]["$ref"] = "#/components/schemas/partition_parameters"
+    openapi_schema["paths"]["/general/v0/general"]["post"]["responses"]["200"]["content"][
+        "application/json"
+    ]["schema"] = {"$ref": "#/components/schemas/Elements"}
+
+    # Schema changes
+
+    # Add securitySchemes
+    # TODO: Implement security per the FastAPI documentation:
+    # https://fastapi.tiangolo.com/reference/security/?h=apikey
+    openapi_schema["components"]["securitySchemes"] = {
+        "ApiKeyAuth": {
+            "type": "apiKey",
+            "name": "unstructured-api-key",
+            "in": "header",
+            "x-speakeasy-example": "YOUR_API_KEY",
+        }
+    }
+
+    # TODO: Instead of a list of paramaters, crete a PartitionParameters model
+    # and declare schema keys (type, format, description) as attributes
+    # https://fastapi.tiangolo.com/reference/openapi/models/?h=model
+    # Update the schema key from `Body_partition` to `partition_paramaters`
+
+    # Must manually correct the schema for the files parameter as due to a bug
+    # described here: https://github.com/tiangolo/fastapi/discussions/10280
+    # files parameter cannot be described with an annotation.
+    # TODO: Check if the bug is fixed and remove this workaround
+    for key in openapi_schema["components"]["schemas"]:
+        if "Body_general_partition_general_v0_general_post" in key:
```

Comment:
That's something hard to avoid with app-defined parameters.

Reply:
I suppose we can dig into later! It will already be a huge improvement to the swagger docs, so not too worried about the naming here yet.